transport.go

```diff
@@ -57,7 +58,24 @@ func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	if err := req.Write(t.conn); err != nil {
 		return nil, err
 	}
-	return http.ReadResponse(t.reader, req)
+	var resp2 *http.Response
+	var err2 error
+
+	resp2,err2 = http.ReadResponse(t.reader, req)
```

Comment:
I'm not sure it's correct to read another response here. This function gets called by [ProxyHandler.proxyRequest()](https://github.com/samuong/alpaca/blob/4e40b40f48286fce494e9ec752915e64d3009f94/proxy.go#L190), and the body of the request gets closed [later in that file](https://github.com/samuong/alpaca/blob/4e40b40f48286fce494e9ec752915e64d3009f94/proxy.go#L206). Since this is done in a `defer` statement, which I think is the bug that you're running into.

If that's the case, then I think #120 would fix it - any chance you can give that branch a try and let me know if it fixes your issue?

Reply:
sure let me test it out and come back to you, most likely removing defer will also fix the issue